pkg/pod/status.go

```diff
@@ -210,9 +236,12 @@ func setTaskRunStatusBasedOnStepStatus(ctx context.Context, logger *zap.SugaredL
 					merr = multierror.Append(merr, err)
 				}
 
-				taskResults, filteredResults := filterResults(results, specResults)
+				taskResults, filteredResults := filterResults(results, specResults, spireEnabled)
 				if tr.IsDone() {
 					trs.TaskRunResults = append(trs.TaskRunResults, taskResults...)
+					if spireEnabled {
```

Comment:
You might need to repeat this line after line 211 as well for status update after fetching results from sidecar logs.

Reply:
Thats right. But signing itself is not done for results from sidecar logs. As such if we verify here, it will surely fail. 
Results from sidecarlogs will be handled in a separate PR after this PR is done.